starknet_py/tests/e2e/account/account_client_test.py

```diff
@@ -125,6 +127,47 @@ async def test_estimate_fee_for_declare_transaction(
     )
 
 
+@pytest.mark.asyncio
+async def test_estimate_fee_for_declare_transactions(
+    new_gateway_account_client, map_compiled_contract, erc20_compiled_contract
+):
+    # TODO: extend this test to new_account_client
+    declare_tx_map = await new_gateway_account_client.sign_declare_transaction(
+        compiled_contract=map_compiled_contract, max_fee=MAX_FEE
+    )
+
+    compiled_contract = create_compiled_contract(
+        compiled_contract=erc20_compiled_contract
+    )
+    new_nonce = declare_tx_map.nonce + 1
+    declare_tx = Declare(
+        contract_class=compiled_contract,
+        sender_address=new_gateway_account_client.address,
+        max_fee=0,
+        signature=[],
+        nonce=new_nonce,
+        version=new_gateway_account_client.supported_tx_version,
+    )
+
+    max_fee = await new_gateway_account_client._get_max_fee(  # pylint: disable=protected-access
+        transaction=declare_tx, max_fee=MAX_FEE
+    )
+    declare_tx = dataclasses.replace(declare_tx, max_fee=max_fee)
```

Comment:
Why do we need to add `max_fee` here?

Reply:
It's the internals of `account_client.sign_declare_tx` brought a level up, to set a nonce that is following the first declares transaction, I'll wrap it in a function in a moment.